File: src/Z.Core/System.Object/Casting/Object.IsAssignableFrom.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;

public static partial class Extensions
{
    /// <summary>
    ///     An object extension method that query if '@this' is assignable from.
    /// </summary>
    /// <typeparam name="T">Generic type parameter.</typeparam>
    /// <param name="this">The @this to act on.</param>
    /// <returns>true if assignable from, false if not.</returns>
    public static bool IsAssignableFrom<T>(this object @this)
    {
        Type type = @this.GetType();
        return type.IsAssignableFrom(typeof (T));
    }

    /// <summary>
    ///     An object extension method that query if '@this' is assignable from.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="targetType">Type of the target.</param>
    /// <returns>true if assignable from, false if not.</returns>
    public static bool IsAssignableFrom(this object @this, Type targetType)
    {
        Type type = @this.GetType();
        return type.IsAssignableFrom(targetType);
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.Object/Casting/Object.IsAssignableFrom.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library


Public Module Extensions_330

	''' <summary>
	'''     An object extension method that query if '@this' is assignable from.
	''' </summary>
	''' <typeparam name="T">Generic type parameter.</typeparam>
	''' <param name="this">The @this to act on.</param>
	''' <returns>true if assignable from, false if not.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function IsAssignableFrom(Of T)(this As Object) As Boolean
		Dim type As Type = this.[GetType]()
		Return type.IsAssignableFrom(GetType(T))
	End Function

	''' <summary>
	'''     An object extension method that query if '@this' is assignable from.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="targetType">Type of the target.</param>
	''' <returns>true if assignable from, false if not.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function IsAssignableFrom(this As Object, targetType As Type) As Boolean
		Dim type As Type = this.[GetType]()
		Return type.IsAssignableFrom(targetType)
	End Function
End Module


